consensus/types/src/test_utils/test_random/bitfield.rs

```diff
@@ -4,8 +4,24 @@ use smallvec::smallvec;
 
 impl<N: Unsigned + Clone> TestRandom for BitList<N> {
     fn random_for_test(rng: &mut impl RngCore) -> Self {
-        let mut raw_bytes = smallvec![0; std::cmp::max(1, (N::to_usize() + 7) / 8)];
+        let initial_len = std::cmp::max(1, (N::to_usize() + 7) / 8);
+        let mut raw_bytes = smallvec![0; initial_len];
         rng.fill_bytes(&mut raw_bytes);
+
+        let highest_set_bit = raw_bytes
+            .iter()
+            .enumerate()
+            .rev()
+            .find(|(_, byte)| **byte > 0)
+            .map(|(i, byte)| i * 8 + 7 - byte.leading_zeros() as usize)
+            .unwrap_or(0);
+
+        let actual_len = highest_set_bit / 8 + 1;
+
+        if actual_len < initial_len {
+            raw_bytes.truncate(actual_len);
+        }
+
```

Comment:
the issue here was we were trying to create bit lists from invalid SSZ bytes because the bytes were random. We aren't supposed to have trailing zeros in the bytes representation but it should be possible still that the actual bit field length is more than 8 bits less than `N`. I tried to make this simpler because I think we don't actually need to know the last non-zero bit, just the last non-zero byte